lib/src/homogeneous.rs

```diff
@@ -41,6 +41,19 @@ impl ProcessorBase for HomogeneousProcessor {
     fn suspend_execution(&mut self, core_id: usize) -> Option<NodeData> {
         self.cores[core_id].suspend_execution()
     }
+
+    fn get_max_value_index(&self, key: &str) -> Option<usize> {
+        self.cores
+            .iter()
+            .enumerate()
+            .filter_map(|(index, core)| {
+                core.processing_node()
+                    .as_ref()
+                    .and_then(|node_data| node_data.params.get(key).map(|&value| (index, value)))
+            })
+            .max_by_key(|&(_, value)| value)
+            .map(|(index, _)| index)
```

Comment:
Fixed a few clarifications.

```suggestion
        self.cores.iter()
    .enumerate()
    .filter_map(|(index, core)| {
        let node_data = core.processing_node().as_ref()?;
        let value = node_data.params.get(key)?;
        Some((index, *value))
    })
    .max_by_key(|&(_, value)| value)
    .map(|(index, _)| index)
```

Reply:
Changed.